File: getsurei.py
# Author: nakinor
# Created: 2011-04-15
# Revised: 2016-03-19

# 月齢を計算して求めるプログラム

import datetime

a_date = datetime.datetime.today()
a_dict = {1:0, 2:2, 3:0, 4:2, 5:2, 6:4, 7:5, 8:6, 9:7, 10:8, 11:9, 12:10}
a = ((a_date.year - 11) % 19) * 11
b = a_dict[a_date.month]
c = a_date.day
getsurei = (a + b + c) % 30

print("今日は{0}年{1}月{2}日です。".format(a_date.year,
                                           a_date.month, a_date.day), end="")
print("月齢は約{0}日です。".format(getsurei))

File: getsurei.fs
(*
   Author: nakinor
   Created: 2015-09-21
   Revised: 2015-09-21

   月齢を計算して求めるプログラム
*)

(* 日付を取得 *)
open System;;
let date = DateTime.Today;;

(* 配列を作っておく リストではなく配列の方で *)
let a_array = [| 0; 2; 0; 2; 2; 4; 5; 6; 7; 8; 9; 10 |];;

(* 計算 *)
let a = ((date.Year - 11) % 19) * 11;;
let b = a_array.[date.Month - 1];;
let c = date.Day;;
let getsurei = (a + b + c) % 30;;

(* 表示 *)
Printf.printf "今日は%d年%d月%d日です。\n月齢は約%d日です。\n"
  date.Year date.Month date.Day getsurei;;
